lodash.src.js

```diff
@@ -7337,6 +7337,46 @@
       };
     }
 
+
+    /**
+     * Creates a function that runs each argument through a transform function.
+     *
+     * @static
+     * @memberOf _
+     * @category Function
+     * @param {Function} func The function to wrap.
+     * @param {...Function} [transforms] The function to transform the corresponding argument.
+     * @returns {Function} Returns the new function.
+     * @example
+     *
+     * var square = function(x) {
+     *   return x * x;
+     * };
+     *
+     * var double = function(x) {
+     *   return x * 2;
+     * };
+     *
+     * var fn = _.useWith(function(x, y) {
+     *   return [x , y];
+     * }, square, double);
+     *
+     * fn(1, 2); // => [1, 4]
+     * fn(5, 10); // => [25, 20]
+     */
+    var useWith = restParam(function(callback, transforms) {
+      return function() {
+        var length = transforms.length,
+            index = -1;
+
+        while (++index < length) {
+          arguments[index] = transforms[index].call(this, arguments[index]);
```

Comment:
@jdalton I don't think we can get around the `apply` since the there can be `n` number of arguments.

@d10 I thought about that. The other solution was convert arguments to an array and modify the new list of arguments, but I don't know which one would cause more overhead.